lib/util/http_utils.dart

```diff
@@ -192,13 +192,22 @@ class HttpUtils {
 
 /// a wrapper class around the http Response
 class Response {
+  bool isLoading = false;
```

Comment:
why not just have an enum instead of 3 variblaes?

Reply:
Changed it to enum